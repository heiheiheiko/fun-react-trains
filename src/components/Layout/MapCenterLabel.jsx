import React from 'react';
import { connect } from "react-redux";
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl'

function MapCenterLabel(props) {
  return (
    <a 
      className="f-row f-ccenter" 
      href={`http://www.google.com/maps/place/${props.mapCenterLat},${props.mapCenterLong}`}
      target="_blank"
      rel="noreferrer noopener"
    >
      <div className="is-size-5 mx-small has-text-weight-semibold"> 
        <FormattedMessage id="aside.mapCenterLabel.latitude"/>: {props.mapCenterLat.toFixed(2)}... 
      </div>
      <div className="is-size-5 mx-small has-text-weight-semibold"> 
        <FormattedMessage id="aside.mapCenterLabel.longitude"/>: {props.mapCenterLong.toFixed(2)}... 
      </div>
    </a>
  );
}

MapCenterLabel.propTypes = {
  mapCenterLat: PropTypes.number,
  mapCenterLong: PropTypes.number,
};

const mapState = state => ({ 
  mapCenterLat: state.mapCenter[1],
  mapCenterLong: state.mapCenter[0],
});

const ConnectedComponent = connect(mapState, null)(MapCenterLabel);
export default ConnectedComponent;
