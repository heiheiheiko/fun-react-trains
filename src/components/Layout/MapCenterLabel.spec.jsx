import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer'
import SpecContainer from '../../../__specs__/helpers/SpecReduxIntelContainer'

import MapCenterLabel from './MapCenterLabel';

describe('<MapCenterLabel>', () => {
  const initialState = { mapCenter: [1,1] };
  const mockStore = configureStore();
  let store, componentContainer;

  beforeEach(() => {
    store = mockStore(initialState)
    componentContainer = <SpecContainer store={store}> <MapCenterLabel /> </SpecContainer>;
  })

  it('could be renderd', () => {
    expect(mount(componentContainer).find(MapCenterLabel).length).toEqual(1)
  });

  it('matched the snapshot', () => {
    const renderedValue =  renderer.create(componentContainer).toJSON()
    expect(renderedValue).toMatchSnapshot();
  });
});