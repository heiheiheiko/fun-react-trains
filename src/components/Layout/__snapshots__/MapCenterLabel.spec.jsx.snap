// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapCenterLabel> matched the snapshot 1`] = `
<div>
   
   
  <a
    className="f-row f-ccenter"
    href="http://www.google.com/maps/place/1,1"
    rel="noreferrer noopener"
    target="_blank"
  >
    <div
      className="is-size-5 mx-small has-text-weight-semibold"
    >
      <span>
        Lat
      </span>
      : 
      1.00
      ...
    </div>
    <div
      className="is-size-5 mx-small has-text-weight-semibold"
    >
      <span>
        Long
      </span>
      : 
      1.00
      ...
    </div>
  </a>
   
   
</div>
`;
